src/main.cpp

```diff
@@ -1221,15 +1221,6 @@ bool AcceptToMemoryPoolWorker(CTxMemPool &pool,
             CCoinsViewMemPool viewMemPool(pcoinsTip, pool);
             view.SetBackend(viewMemPool);
 
-            // do we already have it?
-            bool fHadTxInCache = pcoinsTip->HaveCoinsInCache(hash);
-            if (view.HaveCoins(hash))
-            {
-                if (!fHadTxInCache)
-                    vHashTxnToUncache.push_back(hash);
-                return state.Invalid(false, REJECT_ALREADY_KNOWN, "txn-already-known");
-            }
-
```

Comment:
I think that this is an optimization.  it basically says "if the tx is already in the utxo cache, then clearly we already know about it".  IDK why you'd subsequently want to uncache the tx though... anyway I am fine with leaving it removed